packages/cli/src/commands/worker.ts

```diff
@@ -267,9 +267,9 @@ export class Worker extends BaseCommand {
 	}
 
 	async init() {
-		const configuredShutdownTimeout = config.getEnv('queue.bull.gracefulShutdownTimeout');
-		if (configuredShutdownTimeout) {
-			this.gracefulShutdownTimeoutInS = configuredShutdownTimeout;
+		const shutdownTimeoutFromDeprecatedEnv = config.getEnv('queue.bull.gracefulShutdownTimeout');
+		if (shutdownTimeoutFromDeprecatedEnv !== config.default('queue.bull.gracefulShutdownTimeout')) {
```

Comment:
I think the correct way to check this would be to check for existence of `QUEUE_WORKER_TIMEOUT` env variable